elmslie-core/src/main/java/vivid/money/elmslie/core/store/linking/LinkingExt.kt

```diff
@@ -0,0 +1,69 @@
+package vivid.money.elmslie.core.store.linking
+
+import vivid.money.elmslie.core.store.Store
+
+fun <
+    ParentEvent : Any,
+    ParentState : Any,
+    ParentEffect : Any,
+    ChildEvent : Any,
+    ChildState : Any,
+    ChildEffect : Any,
+> Store<ParentEvent, ParentState, ParentEffect>.linkTo(
+    childStore: Store<ChildEvent, ChildEffect, ChildState>,
+    effectMapper: (ChildEffect) -> ParentEvent?,
+    stateMapper: (ChildState) -> ParentEvent?,
+) = apply {
+    launch {
+        childStore.states().collect { stateMapper.invoke(it)?.let(this@linkTo::accept) }
+        childStore.effects().collect { effectMapper.invoke(it)?.let(this@linkTo::accept) }
+    }
+}
+
+fun <
+    ParentEvent : Any,
+    ParentState : Any,
+    ParentEffect : Any,
+    ChildEvent : Any,
+    ChildState : Any,
+    ChildEffect : Any,
+> Store<ParentEvent, ParentState, ParentEffect>.linkTo(
+    childStore: Store<ChildEvent, ChildEffect, ChildState>,
+    effectMapper: (ChildState, ChildEffect) -> ParentEvent? = { _, _ -> null },
+    stateMapper: (ChildState, ChildState) -> ParentEvent? = { _, _ -> null },
+) = apply {
+    launch {
+        var currentState = childStore.currentState
+        childStore.states().collect {
+            stateMapper.invoke(currentState, it)?.let(this@linkTo::accept)
+            currentState = it
+        }
+        childStore.effects().collect {
+            effectMapper.invoke(currentState, it)?.let(this@linkTo::accept)
```

Comment:
For what purpose you keep local variable [currentState] and pass it for effect mapper?

Reply:
I thought about these extensions again. And I'm pretty sure that we should stay all of them inside the library. Let's allow to choose what extension to use for library users.

Reply:
But maybe we should have only linkListTo implementation, and other extensions should call linkListTo extension (of course map single elements to listOf(element)). What do you think?